File: javascript/backend-axios/08-empty-subdivision.mjs
// Set your CountriesDB private API key here
const API_KEY = 'your_private_api_key_here';

import axios from 'axios';

const API_BASE = 'https://api.countriesdb.com';

if (!API_KEY || API_KEY === 'your_private_api_key_here') {
  throw new Error('Please set your COUNTRIESDB_PRIVATE_KEY in the script');
}

const client = axios.create({
  baseURL: API_BASE,
  headers: {
    Authorization: `Bearer ${API_KEY}`,
    'Content-Type': 'application/json',
  },
});

// 8. Empty subdivision validation (for countries with no subdivisions like Antarctica)
async function main() {
  try {
    console.log('Validating empty subdivision code for country: AQ');
    const { data } = await client.post('/api/validate/subdivision', {
      country: 'AQ',
      code: '',
    });
    console.log('Response:', JSON.stringify(data, null, 2));
    
    if (!data.valid) {
      throw new Error('Empty subdivision validation failed');
    }
    
    console.log('Empty subdivision is valid (country has no subdivisions)');
  } catch (err) {
    console.error('Validation failed:', err.response?.data ?? err.message);
    process.exit(1);
  }
}

main();

File: ruby/backend-faraday/08-empty-subdivision.rb
require 'bundler/setup'
require 'faraday'
require 'json'

# Set your CountriesDB private API key here
api_key = 'YOUR_API_KEY' # Replace with your actual API key
if api_key == 'YOUR_API_KEY'
  abort 'Please set your API_KEY in the script'
end

client = Faraday.new(
  url: 'https://api.countriesdb.com',
  headers: {
    'Authorization' => "Bearer #{api_key}",
    'Content-Type' => 'application/json'
  }
)

def call(client, path, payload)
  response = client.post(path, payload.to_json)
  body = JSON.parse(response.body)
  raise "Request failed: #{response.status} #{body}" unless response.success?

  body
end

# 8. Empty subdivision validation (for countries with no subdivisions like Antarctica)
begin
  puts 'Validating empty subdivision code for country: AQ'
  result = call(client, '/api/validate/subdivision', { country: 'AQ', code: '' })
  puts "Response: #{JSON.pretty_generate(result)}"
  
  unless result['valid']
    raise 'Empty subdivision validation failed'
  end
  
  puts '✅ Empty subdivision is valid (country has no subdivisions)'
rescue => e
  puts "❌ Validation failed: #{e.message}"
  exit 1
end

